assets/js/theme/global/modal.js

```diff
@@ -196,12 +213,89 @@ export class Modal {
         this.$content.html('');
     }
 
+    setupFocusableElements(modalType) {
+        this.$preModalFocusedEl = $(document.activeElement);
+        
+        const $collection = focusableElements[modalType]();
+
+        this.findElementToFocus($collection);
+
+        const onTabbingWithArgs = (event) => this.onTabbing(event, modalType, this.findElementToFocus.bind(this));
+
+        $('#modal').on('keydown', onTabbingWithArgs);
+    }
+
+    findElementToFocus($candidatesCollection, $reserveCollection = null) {
+        $candidatesCollection.each((index, element) => {
+            const $focusCanditate = $(element);
+            $focusCanditate.focus();
+            if ($focusCanditate.is($(document.activeElement))) {
+                return false;
+            };
+            // if no appropriate candidate - find appropriate in full collection
+            if (index === $candidatesCollection.length - 1 && $reserveCollection) {
+                this.findElementToFocus($reserveCollection);
+            };
+        });
+    }
+
+    onTabbing(event, modalType, findElementToFocus) {
```

Comment:
Why do we need to pass class method in this case? And currying here is still a better idea https://learn.javascript.ru/currying-partials